src/Nethermind/Nethermind.JsonRpc/Modules/Eth/GasPriceOracle.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Int256;
+
+namespace Nethermind.JsonRpc.Modules.Eth
+{
+    public class GasPriceOracle : IGasPriceOracle
+    {
+        public UInt256? FallbackGasPrice { get; private set; }
+        public List<UInt256> TxGasPriceList { get; private set; }
+        protected UInt256? LastGasPrice { get; private set; }
+        private Block? LastHeadBlock { get; set; }
+        private readonly UInt256? _ignoreUnder;
+        private readonly int _blockLimit;
+        private readonly int _softTxThreshold;
+        private readonly UInt256 _baseFee;
+        private readonly ITxInsertionManager _txInsertionManager;
+        private readonly IHeadBlockChangeManager _headBlockChangeManager;
+
+        public GasPriceOracle(
+            ISpecProvider specProvider,
+            UInt256? ignoreUnder = null, 
+            int? blockLimit = null, 
+            UInt256? baseFee = null,
```

Comment:
Could you explain why do you pass baseFee?

Reply:
I remembered for EIP1559, the baseFee was supposed to change according to the fullness of the block and I thought that I should start with the current baseFee and update it.

But after you asked that, I realized that this function is not responsible for calculating new baseFees; we are calculating the new EffectiveGasPrice to pay for transactions in the next block.